### DISCUSSION TITLE
useI18n() must be called at the top of a `setup` function (in pinia store)

### DISCUSSION THREAD

**Discussion by yxyx-github** [quasarframework/quasar#15185] [General - Components / Directives / etc]
Good evening everyone,

I have an issue with Pinia stores and Vue Router.
When I use useI18n inside a Pinia store (composition api) and access the store in Router.beforeEach() at the same time, it cause the following error:

```
vue-router.mjs:3451 SyntaxError: Must be called at the top of a `setup` function (at message-compiler.esm-bundler.js:54:19)
    at createCompileError (message-compiler.esm-bundler.js:54:19)
    at createI18nError (vue-i18n.esm-bundler.js:100:12)
    at useI18n (vue-i18n.esm-bundler.js:2214:15)
    at example-store.ts:5:16
    at pinia.mjs:1436:32
    at EffectScope.run (reactivity.esm-bundler.js:34:24)
    at pinia.mjs:1436:22
    at EffectScope.run (reactivity.esm-bundler.js:34:24)
    at createSetupStore (pinia.mjs:1434:33)
    at useStore (pinia.mjs:1705:17)
```

Reproduction repository: [https://github.com/yxyx-github/pinia-use-i18n-issue](https://github.com/yxyx-github/pinia-use-i18n-issue)

Do you know why this happens?

As an alternate in the store I could do:
```import i18n from 'src/locale/index'```
and then:
```i18n.global.t```
but I'm still confused about this. It would be great if someone of you could explain to me, why this error happens.

Thanks in advance.

Best regards
yxyx-github

**Comment by MrNightingale** [comment]
Experienced the same issue in Nuxt. As a workaround I've imported useNuxtApp in store.

```
import { useNuxtApp } from '#imports'

export const fooStore = defineStore('foo', () => {
  const { $i18n } = useNuxtApp()
  const { t } = $i18n
  ...
}
```

**Comment by rstoenescu** [comment]
Reproduction repo no longer available... Closing.
